aea/crypto/ethereum_base.py

```diff
@@ -0,0 +1,105 @@
+#   See the License for the specific language governing permissions and
+#   limitations under the License.
+#
+# ------------------------------------------------------------------------------
+
+"""Module wrapping the public and private key generation for interacting with ethereum."""
+
+from typing import Optional
+import logging
+
+from eth_account.messages import encode_defunct  # type: ignore
+from web3 import Web3       # type: ignore
+from eth_account import Account     # type: ignore
+from eth_keys import keys       # type: ignore
+from pathlib import Path
+
+logger = logging.getLogger(__name__)
+
+
+class EthCryptoError(Exception):
+    """Exception to be thrown when cryptographic signatures don't match!."""
+
+
+class EthCrypto(object):
+    """Class wrapping the Entity Generation from Fetch.AI ledger."""
+
+    def __init__(self, private_key_path: Optional[str] = None):
+        """Instantiate a crypto object."""
+        self.account = self._generate_private_key() if private_key_path is None else self._load_private_key_from_path(private_key_path)
+        self._display_address = self.account.address
+        self._bytesRepresentation = Web3.toBytes(hexstr=self.account.privateKey.hex())
+        self._public_key = keys.PrivateKey(self._bytesRepresentation).public_key
+
+    @property
+    def public_key(self) -> str:
+        """
+        Return a public key in hex format.
+
+        :return: a public key string in hex format
+        """
+        return self._public_key
+
+    @property
+    def display_address(self) -> str:
+        """
+        Return the display_address for the key pair.
+
+        :return: a display_address str
+        """
+        return str(self._display_address)
+
+    def _load_private_key_from_path(self, file_name) -> Account:
+        """
+        Load a private key in hex format from a file.
+
+        :param path: the path to the hex file.
+
+        :return: the Entity.
+        """
+        path = Path(file_name)
+        try:
+            if path.is_file():
+                with open(path, "r") as key:
+                    data = key.read()
+                    account = Account.from_key(data)
+            else:
+                account = self._generate_private_key()
+            return account
+        except IOError as e:
+            logger.exception(str(e))
+
+    def sign_transaction(self, message: str) -> bytes:
+        """
+        Sing a transaction to send it to the ledger.
+
+        :param message:
+        :return: Signed message in bytes
+        """
+        m_message = encode_defunct(text=message)
+        signature = self.account.sign_message(m_message)
+        return signature
+
+    def _generate_private_key(self) -> Account:
+        """Generate a key pair for ethereum network."""
+        path = Path("eth_pk.txt")
+        print(path)
+        pk = Account.create()
+        with open(path, "w+") as file:
```

Comment:
what if a private key already exists?

Reply:
You just load it and you create the account.

Reply:
See above. Overwrite issue.